src/datascience-ui/interactive-common/variableExplorer.tsx

```diff
@@ -213,6 +218,12 @@ export class VariableExplorer extends React.Component<IVariableExplorerProps, IV
     private renderGrid() {
         const newGridHeight = this.calculateGridHeight(this.props.viewHeight);
 
+        // Customize our empty rows views based on if we are in the act of requesting variables
+        // Allows us to say "Loading" initially versus "No Variables Defined"
+        // We have to handle this with two different React components as the grid emptyRowsView takes a component
+        // not an element as a property and internally calls createElement without properties for the view
+        const emptyRowsView = this.props.requestInProgress ? VariableExplorerLoadingRowsView : VariableExplorerEmptyRowsView;
```

Comment:
Spent a bit of time trying to put both views into one Component. But this prop takes a Component class (not an element like the rowRenderer) that is directly created internally in Adazzle with createElement and without any props. So this seemed like the actual clean way to do this.

We only do this for the empty rows view, because if we already have rows we just want to update, not flash to Loading back and forth while running new cells.